pulp_ansible/app/tasks/collectionversion_index.py

```diff
@@ -0,0 +1,188 @@
+
+    if is_latest:
+        repo_v = None
+    else:
+        repo_v = repository_version
+
+    if not _is_indexable(repository, repo_v):
+        return
+
+    cvs = repository_version.content.filter(pulp_type="ansible.collection_version")
+    cvs = [x.cast() for x in cvs]
+
+    # get the set of signatures in this repo version
+    repo_signatures = repository_version.content.filter(pulp_type="ansible.collection_signature")
+    repo_signatures = [x.cast() for x in repo_signatures]
+
+    # get the set of deprecations in this repo version
+    deprecations = repository_version.content.filter(pulp_type="ansible.collection_deprecation")
+    deprecations = [x.cast() for x in deprecations]
+    deprecations_set = set([(x.namespace, x.name) for x in deprecations])
+
+    namespaces = repository_version.content.filter(pulp_type="ansible.namespace")
+    namespaces = [x.cast() for x in namespaces]
+    namespaces = {x.name: x for x in namespaces}
+
+    # map out the namespace(s).name(s) for everything in the repo version
+    colset = set([(x.namespace, x.name) for x in cvs])
+
+    log.info("Repository collections ...")
+    log.info(colset)
+
+    # clean up any repo+namespace+name indexes that no longer live in the repo ...
+    cvindexes = CVIndex.objects.filter(repository=repository, repository_version=repo_v)
+    for cvi in cvindexes:
+        if (cvi.collection_version.namespace, cvi.collection_version.name) not in colset:
+            log.info(
+                (
+                    f"DELETE {cvi.repository.name}"
+                    + f".{cvi.collection_version.namespace}"
+                    + f".{cvi.collection_version.name}"
+                    + f".{cvi.collection_version.version} FROM INDEX"
+                )
+            )
+            cvi.delete()
+            continue
+        log.info(
+            (
+                f"KEEP {cvi.repository.name}"
+                + f".{cvi.collection_version.namespace}"
+                + f".{cvi.collection_version.name}"
+                + f".{cvi.collection_version.version} IN INDEX"
+            )
+        )
+
+    # iterate through each collection in the repository
+    for colkey in colset:
+        namespace = colkey[0]
+        name = colkey[1]
+
+        # get all the versions for this collection
+        related_cvs = [x for x in cvs if x.namespace == namespace and x.name == name]
+
+        # compute the highest version (allow pre-releases too?)
+        versions = [x.version for x in related_cvs]
+
+        versions = sorted(versions, reverse=True, key=lambda x: Version(x))
+        highest_version = versions[0]
+
+        # clean up any CV indexes with a version no longer in the repo ...
+        cvindexes = CVIndex.objects.filter(
+            repository=repository,
+            collection_version__namespace=namespace,
+            collection_version__name=name,
+        )
+        for cvi in cvindexes:
+            if cvi.collection_version.version not in versions:
```

Comment:
I think you need to convert to a Version object first.
```suggestion
            if Version(cvi.collection_version.version) not in versions:
```